reference/src/layout/structs-and-tuples.md

```diff
@@ -122,6 +122,27 @@ compiler will not reorder it, to allow for the possibility of
 unsizing. E.g., `struct Foo { x: u16, y: u32 }` and `struct Foo<T> {
 x: u16, y: T }` where `T = u32` are not guaranteed to be identical.
 
+#### Default layout of structs with a single non-zero-sized field
+
+The default layout of structs with a single non-zero-sized field is the same as
+the layout of that field if the alignment requirement of all other fields is 1.
```

Comment:
This sentence puts part of the conditions on other fields at the beginning and parts at the end.

IMO we should, somewhere, define a term for a "zero-sized-one-aligned type". Working title ZSOAT. Then we can just say things like "structs with a single non-ZSOAT field".

Reply:
The glossary would be the right place to define that term. We should probably define ZST as zero-sized-type there as well. ZSOAT is also quite a mouthful, maybe just `ZST1` ?

Reply:
With a numeral it makes it easy to talk about higher aligned zero sized types as well, `ZST4` for example

Reply:
ZSOAT was meant as a proposal that is so bad that we clearly cannot use it. :)

`ZST1`/`ZST4` sounds good to me! The general `ZSTN` looks a bit funny though, `ZST-N` would be clearer, so maybe it should be `ZST-1` then?

Reply:
The obvious alternative is `1-ZST`. "One-aligned zero-sized type" works better than "zero-sized type, 1-aligned".

Reply:
> The obvious alternative is 1-ZST. "One-aligned zero-sized type" works better than "zero-sized type, 1-aligned".

I like `1-ZST`.

Reply:
It is unclear whether this shall that be done in a different PR or here. Adding an abbreviation for these types feels like an orthogonal problem to me, and requires going through the whole UCGs to use the abbreviation appropriately.

Reply:
I'd rather add the abbreviation before we land all the PRs you opened that talk about N-ZST. Then we'll have strictly fewer places to fix later.